akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/ActiveActiveEventSourcing.scala

```diff
@@ -92,21 +97,43 @@ object ActiveActiveEventSourcing {
    * A different journal plugin id can be configured using withJournalPluginId after creation. Different databases
    * can be used for each replica.
    * The events from other replicas are read using PersistentQuery.
-   * TODO support a different query plugin id per replicas: https://github.com/akka/akka/issues/29257
    *
    * @param replicaId The unique identity for this entity. The underlying persistence id will include the replica.
    * @param allReplicaIds All replica ids. These need to be known to receive events from all replicas.
-   * @param queryPluginId Used to read the events from other replicas. Must be the query side of your configured journal plugin.
-   * @return
+   * @param queryPluginId A single query plugin used to read the events from other replicas. Must be the query side of your configured journal plugin.
    */
-  def apply[Command, Event, State](
+  def withSharedJournal[Command, Event, State](
       entityId: String,
-      replicaId: String,
-      allReplicaIds: Set[String],
+      replicaId: ReplicaId,
+      allReplicaIds: Set[ReplicaId],
```

Comment:
so the map vs set overloading didn't work here?

Reply:
Not sure what that refers to

Reply:
Ah, yes, got a name clash, I think it is related to having two parameter lists but not sure.